### ISSUE TITLE
Ability to exclude albums from Collection

### ISSUE THREAD

**Issue by noearchimede** [deckerst/aves#520]
First of all (and regardless of this feature request), thank you for developing and maintaining Aves! I have been using it for a while now and I don't think I'm ever going back to any other gallery app.

I have many types of photos on my Android devices. Consider these broad categories:
1. nice pictures of people, places, events, ecc. – saved in various subfolders of DCIM and Pictures
2. pictures of receipts, documents, screenshots, etc. that I may need for a short period of time for a specific reason and then will be deleted – saved directories such as in Pictures/Screenshots, Pictures/myTemporaryPhotos, ...
3. media "generated" by apps, such as WhatsApp photos, PNGs of handwritten notes on my tablet, etc. – saved wherever these apps save them

In general when I scroll through my photos I'd like to only see (1.). But from time to time I also want to look at photos such as (2.) or (3.). If I only ever used Album view there would be no problem, but I'd still like to be able to use the (very beautiful!) "collections" tab to view the "nice photos" (1.) all in one place (regardless of what album they are in or what camera app they were taken with). On the other side Album view is perfect for, say, findind a picture of a specific document or scrolling throught the latest WhatsApp images.

So I would like to have an option to select, for every album, whether it should appear in Collection or not (the default being yes) without hiding it completely (then maybe this "not in collection" status could be marked by some badge in Album view).

I think this behaviour is significantly different from current options:
- hiding an album doesn't have the same effect, because then the media in it is not easily accessible anymore. Going into privacy settings, unhiding it, and then hiding it again is in theory an option but in my opinion too complicated for something that may happen daily. (My current solution is actually to hide those unwanted folders in Aves and use the preinstalled gallery app to browse them when I need to... but I would like to not need two separate gallery apps)
- not hiding any album and only using Album view is also possible, but it is a bit a shame because the Collection view is a wolderful way to scroll through photos
- setting up appropriate search criteria to temporarly hide the same photos (groups 2. and 3. in the example above) every time I want to see my "nice" photos (1.) is also not very convenient

In other terms what I'm asking is if it would be possible to redefine "collection" from "all images on my devices, except permanently hidden ones" to "all images worth scrolling through". And I think that making the choice of whether to include a photo in "collection" based on album might be a sensible, but that is not the only possibility and is not the core of my feature request.

**Comment by deckerst** [comment]
Thanks for the kind words!

Probably not going to implement things as you suggest it. I think it's an interesting use case, but I'm careful not to complexity the app UI for regular users if i can avoid it. The concept of "hiding" is already confusing to some.

But i have a proposal. What if you could quickly switch between a set of settings (including what's hidden and what's not). Let's say we call this set of settings a "profile". You could switch from one profile to another by going to Settings and there would be a switcher in the top right corner (next to the "..." menu. With this, you wouldn't have to go deep in the settings menu just to unhide something. Also, you could have other settings (overlay, viewer, etc.) associated with that profile. Finally, it wouldn't complexify the UI, as the feature is only visible in the Settings app bar.

**Comment by noearchimede** [comment]
Thanks for your reply!

I personally like the concept of settings profiles and I do think that it would be a useful to have that in Aves, as one may use the app for a number of different purposes which may have different requirements (show a friend your vacation pictures? minimal overlays, hide work-related stuff! organizing your photos? show all possible overlays and don't hide any picture!).

I also understand your desire to keep the UI as simple as possible. You're right that having two types of "hiding" would probably confuse a lot of people.

And what would you say about implementing a restriction of the photos shown in Collection directly in Settings? That way a user will never end up having photos that are in Albums but not in Collection unless they navigate to the appropriate Settings pane (let's say "Set up Collection") and select the albums they wish to see in Collections explicitly, at which poit they will not be confused because they will know about the "Set up Collection" setting. Would that be feasable or still be too similar to the "hide" feature in your opinion?

**Comment by deckerst** [comment]
> That way a user will never end up having photos that are in Albums but not in Collection

Currently albums are derived from images and videos in the collection. So it's always consistent, and I don't understand the issue you refer to.

**Comment by noearchimede** [comment]
Sorry I wasn't clear, I was referring to the fact that if a way to exclude some albums from Collections was implemented into the main UI (I mean outside of Settings) then a use might use it unadvertently and then be confused about what appened (let's say exclude the DCIM/Camera folder and then wonder why new photos don't appear in collections).

I was just reflecting on your remark in your previous post that you are "careful not to complexity the app UI for regular users if [you] can avoid it."

**Comment by pancito-uwu** [comment]
I think this could be achieved in a similar way as how the VLC app does it, where you can simply choose which directories to look into.
I also thought that maybe the app could ignore folders that contain a specific file, like, if there's a file called `collectionsignore.aves` or something like that then any media from that directory won't appear in Collections, and similarly for albums.

**Comment by andrej-dyck** [comment]
Hello,

I just found the app and I like it a lot. 

I have a similar need like the author of this issue: I'd like to use the "collections" view as default, but with some albums are excluded (either by creating a collection with selected albums or excluding albums from "all"). This is so that I can view photos and not other pictures like screenshots, downloaded pictures, or photos of recipes. 

But I'd also like to access those albums, so the "hide" option is not suitable for this use case (but the option is still necessary for excluding paths globally for the app).

Is there any progress on a solution for this? Any help needed?

Kind regards,
Andrej

**Comment by deckerst** [comment]
Do you think what i suggested in [this comment](https://github.com/deckerst/aves/issues/520#issuecomment-1424452850) would be a good fit for your use case?

In the meantime, I've recently added toggles to hidden items in the settings, which make it easier to show them for a while without removing the filters from settings completely.

**Comment by andrej-dyck** [comment]
Hi @deckerst ,

thanks. To be honest, I don't think that introducing another concept like "profile" is a good idea. Most photographers have a good understanding of what collections are. So, I'd be in favor of somehow incorporating those into a solution.

If you're concerned about complicating the user-facing UI, I think a setting to configure the collection is viable through the settings menu. 

Could you reuse / amend the "hide" option? E.g., split it into a choice: "exclude from ..." and then "app" or "collection".

**Comment by andrej-dyck** [comment]
Another idea: the collection already has a powerful filtering. I think it would cover most use cases if one could save a "default" filter or the last used filter is persistent through app restart (a setting probably).

What do you think?

**Comment by deckerst** [comment]
I don't know what "Collection" specifically means for photographers. Here in Aves it *does* mean all media displayed by the app (without the hidden items), with or without additional filters (like videos, favorites, etc.)

Back to your use case, if the idea is to open the app to a `Collection` page with your whole collection minus some albums, it's already possible. Not elegant, but possible: 
1) go to the `Collection` page without any filter
2) tap on the top to get the `Search` page 
3) type the name of an album you want excluded
4) long press on the filter and select "filter out"
*Repeat 2 3 4 for all the Albums you want to exclude*
5) in the Collection page, select in the top right menu "Set as Home"

**Comment by andrej-dyck** [comment]
That's a solution, thanks. I find "set as home" confusing here, but it works. 

Is there an option to hide filter specifics for "home"; e.g., where the filter items are "collapsed" and don't take up valuable screen space?

---

Regarding the topic of terminology: what Aves calls "collections" is rather known as "library", "catalogue", or simply "all photos". Here are descriptions of "collections" in two of the most popular tools:
- Lightroom https://www.lightroomqueen.com/collections-organize-photos/
- CaptureOne https://support.captureone.com/hc/en-us/articles/360002554597-Adding-images-to-Capture-Collections-Catalogs
*tl;dr*: Collections are user-defined virtual folders where one photo can be in many (or none) collections. They are either defined by manual inclusion of photos of through filters.

**Comment by deckerst** [comment]
> Is there an option to hide filter specifics for "home"; e.g., where the filter items are "collapsed" and don't take up valuable screen space?

I don't understand. Could you take/annotate a screenshot, if you're talking about UI?

As for the terminology, i didn't want to use something like "all photos", because the collection page can show filtered or unfiltered items. "Library" would work, though. 

For "collection" as virtual album, a possible future feature is that kind of virtual album, but for now i wanted to keep it simple and equate "album" with "folder" because I've noticed general users being confused by apps providing advanced albums like that.

**Comment by andrej-dyck** [comment]
> I don't understand. Could you take/annotate a screenshot, if you're talking about UI?

Certainly. 
![IMG_20240407_195038.jpg](https://github.com/deckerst/aves/assets/43913051/4def8cee-b916-4302-8b5d-88c311c4ef10)

The filters in the red rectangle take away a lot of vertical space. If it's a default filter ("set as home"), they are then always there. So, a possibility to hide those filters (if they are the default) to show more content would be preferable. 

An indication that a filter is applied (e.g., like the red dot I added) is an addition.

---

> For "collection" as virtual album, a possible future feature is that kind of virtual album, but for now i wanted to keep it simple and equate "album" with "folder" because I've noticed general users being confused by apps providing advanced albums like that.

The app basically has it with the filters on "collection". If it's now possible to save and name a filter (like the present ones in the settings), and then pin them as navigation items, it's (mostly?) done.

**Comment by deckerst** [comment]
Ok, i understand what you request, but i disagree with you that "If it's a default filter ("set as home"), they are then always there", because "home" is only a starting point. You can always tap on the filters to remove them (or reverse them). I don't want to hide them as implicit filters simply because they define your home. 

> The app basically has it with the filters on "collection". If it's now possible to save and name a filter (like the present ones in the settings), and then pin them as navigation items, it's (mostly?) done.

True. Feature-wise the code allows for it, but the work here would mostly be on terminology, UX and consistency. It's not necessarily the easiest part as it's all too easy to make the app feel bloated by overwhelming users with arcane terms and niche features.

**Comment by andrej-dyck** [comment]
> [..] i disagree with you that "If it's a default filter ("set as home"), they are then always there", because "home" is only a starting point. You can always tap on the filters to remove them (or reverse them). I don't want to hide them as implicit filters simply because they define your home.

I see your point. I hope you also see mine, where with every app start ("set as home"), I have always the same filteres applied, and thus, I just don't need to see them. 

I strongly agree that this might be a poweruser thing, however, I can imagine that once the app would allow users to give a set of filters a name (aka speaking of "collections") -- e.g., "my photos & videos" (+all, -downloads, -screenshots) and "documents & receipts" (+documents) -- the filters don't have to be visible anymore.

What do you think?

> [..] the work here would mostly be on terminology, UX and consistency.

Yes, I haven't dismissed that. If you think I could be of help to bounce some ideas, feel free to reach out. I have a strong background in SE, UX, and photography.

**Comment by juanriccio** [comment]
I've set Aves as my default gallery app in my Android phone. I like it better than Google Photos.

Aves would became even more useful if it allowed me to show by default only certain "albums" or whatever they're called. Take for instance WhatsApp pictures - they litter my own images and photos. There's hundreds of silly memes and gifs scattered among my own pics/videos, which makes it hard to locate quickly what I'm looking for.

It would be so nice if we could choose some folders to always be shown by default, and other that would only be shown when we press the "show all images, even garbage" button! Please, consider.

**Comment by deckerst** [comment]
@juanriccio not long ago I've added a way to toggle hidden items. So it's easier to have all your memes hidden most of the time. When you need to see your memes, just toggle them in the settings, then toggle them again to hide them.

**Comment by juanriccio** [comment]
@deckerst Thanks for your reply! I can't locate the setting about what to hide, or the option to toggle hidden items. I think the option hide/show all would be very useful on the homescreen.

**Comment by deckerst** [comment]
Settings > Privacy > Hidden items. 
![Screenshot_20240623-130421~2](https://github.com/deckerst/aves/assets/13898333/09efe030-3bf8-4e7b-94a5-6e24d3d228e5)

**Comment by juanriccio** [comment]
Only now noticed. I actually wanted to exclude some albums, and I managed to set up such a filter. Not sure how I did, but now the filter is now there. Thank you! Sorry for not confirming earlier.

**Comment by deckerst** [comment]
> The app basically has it with the filters on "collection". If it's now possible to save and name a filter (like the present ones in the settings), and then pin them as navigation items, it's (mostly?) done.

@andrej-dyck implemented as "dynamic albums", in the next release. It's integrated with the other albums. They can be pinned, searched, added to navigation menu, combined with other filters, etc.

**Comment by andrej-dyck** [comment]
> > The app basically has it with the filters on "collection". If it's now possible to save and name a filter (like the present ones in the settings), and then pin them as navigation items, it's (mostly?) done.
> 
> @andrej-dyck implemented as "dynamic albums", in the next release. It's integrated with the other albums. They can be pinned, searched, added to navigation menu, combined with other filters, etc.

Awesome. Thank you!
